src/components/button/button.tsx

```diff
@@ -1,12 +1,21 @@
-import React, { InputHTMLAttributes } from 'react';
-import { StitcheButton } from './styles';
+import React from 'react';
+import { StitchesButton } from './styles';
 
-export interface ButtonProps extends InputHTMLAttributes<HTMLButtonElement> {
-  label: string;
-}
+type StitchesButtonProps = React.ComponentProps<typeof StitchesButton>;
 
-const Button = ({ label }: ButtonProps) => {
-  return <StitcheButton>{label}</StitcheButton>;
+export type ButtonProps = {
+  label?: string;
+  type: string;
```

Comment:
```suggestion
  type: string;
```

Primary can be default